vllm/model_executor/layers/quantization/utils/w8a8_utils.py

```diff
@@ -161,10 +162,14 @@ def apply_fp8_linear(
         # Note: we pad the input because torch._scaled_mm is more performant
         # for matrices with batch dimension > 16.
         # This could change in the future.
+        # We also don't pad when using torch.compile,
+        # as it breaks with dynamic shapes.
+        config = get_current_vllm_config().compilation_config
```

Comment:
Is this cached? It could be expensive each forward call

Reply:
Yes, in eager mode this will get called on every forward pass, but it will only happen once when compiled. In eager mode there isn't really a better way that's still correct - the only way is to check the config context. I don't think this getter is significant but I haven't measured it.

Reply:
We could passs in a `allow_input_padding` flag? and pass it in? I do think this is annoying though. I think it's woth it to do a quick check for performance regressions on a small model eager mode benchmark with `cutlass_scaled_mm` disabled?

Reply:
I think we'd have to pass that flag through the whole call stack though so I don't think it's worth it. I'll run a small model.